src/api-extension/cart.service.ts

```diff
@@ -476,21 +523,11 @@ export class CartService {
     applicableCoupons: StackableRedeemableResponse[],
   ) {
     productsToAdd.map((productToAdd) => {
-      this.countOrderDiscountDifference(
-        validatedCoupons.order,
-        productToAdd.discount_difference,
-      );
+      this.countOrderDiscountDifference(validatedCoupons.order, 0);
     });
 
     productsToAdd.map((productToAdd) => {
-      applicableCoupons
-        .filter((redeem) => redeem.id === productToAdd.code)
-        .map((redeem) => {
-          this.countOrderDiscountDifference(
-            redeem.order,
-            productToAdd.discount_difference,
-          );
-        });
+      applicableCoupons.filter((redeem) => redeem.id === productToAdd.code);
```

Comment:
This filter probably noting change too.
I suppose that functions countOrderDiscountDifference and handleCartDiscountDifferences could be deleted. But please double check it.